src/lib/pubkey/sphincsplus/sphincsplus_common/sp_fors.cpp

```diff
@@ -34,7 +34,12 @@ std::vector<TreeNodeIndex> fors_message_to_indices(std::span<const uint8_t> mess
 
    for(auto& idx : indices) {
       for(uint32_t i = 0; i < params.a(); ++i, ++offset) {
-         idx ^= (((message[offset >> 3] >> (offset & 0x7)) & 0x1) << i);
+         // This is one of the few places where SPHINCS+ round 3.1 and SLH-DSA differ
+         if(params.is_slh_dsa()) {
+            idx ^= (((message[offset >> 3] >> (~offset & 0x7)) & 0x1) << (params.a() - 1 - i));
+         } else {
+            idx ^= (((message[offset >> 3] >> (offset & 0x7)) & 0x1) << i);
+         }
```

Comment:
Yeah, I think this is overkill. (Same argument as below for `prepare_message`)
Also, sadly, it's `(offset & 0x7)`  (instead of `(~offset & 0x7)`) in the second case.

Reply:
Yeah, I agree.